### ISSUE TITLE
Hostname Route still Routing other modules

### ISSUE THREAD

**Issue by faghihian71** [zendframework/zf2#7605]
dependent to #7596 ticket you labeled my issue as a question not a problem but now my problem is why routing by hostname still route another modules 

i asked many question for this problem in stack overflow but nobody answer it yet 

 i have this route configuration for advertise module :

     'router' => array(
         'routes' => array(
             'adview' => array(
                 'type'    => 'segment',
                 'options' => array(
                     'route'    => '/adview[/:hash][/:title]',
                     'constraints' => array(
                      
                     ),
                     'defaults' => array(
                         'controller' => 'Advertise\Controller\Advertise',
                         'action'     => 'show',
                     ),
                 ),
             ),
     )
    )

and a reseller module whit this route configuration:

     'router' => array(
        'routes' => array(
            'Reseller' => array(
                'type'    => 'Hostname',
                'options' => array(
                    'route'    => 'reseller.mytest.com',
                ),
                'may_terminate' => false,
                'child_routes' => array(
                    'home' => array(
                        'type' => 'Zend\Mvc\Router\Http\Literal',
                        'options' => array(
                            'route'    => '/',
                            'defaults' => array(
                                'controller' => 'Reseller\Controller\Reseller',
                                'action'     => 'index'
                            )
                        )
                    ),
    )
    )

so the problem is reseller.mytest.com/adview  work that we expect to not work because adview isn't in scope of reseller module routing

**Comment by adamlundrigan** [comment]
As was said in #7596 the issue tracker isn't the correct place to ask questions like this.  Post your question on StackOverflow and advertise it over the Twitter  (#zf2) and IRC channels (#zftalk on freenode) and I'm sure someone will help you.

**Comment by adamlundrigan** [comment]
TL;DR: if a route applies only to a specific hostname put it as a child of the appropriate Hostname route.

This is the intended operation of the router.  Routes not "fenced" into a specific hostname by a Hostname route will apply to all hostnames, as the default operation of the router is hostname-agnostic (a ZF2 app could be moved from www.example.com to www.otherdomain.com without needing any configuration change and it "just works").
